### ISSUE TITLE
[snip_env seems not work] even set snip_env to {} will cause error

### ISSUE THREAD

**Issue by hiberabyss** [L3MON4D3/LuaSnip#566]
Will have following error, if snip_env is not nil:

![image](https://user-images.githubusercontent.com/2898670/187693967-8b92a2f4-27e1-485b-970a-1e52a02f8036.png)

**Comment by hiberabyss** [comment]
Want to add postfix to setup_env, could this be default?

**Comment by L3MON4D3** [comment]
Ouh, looks like the content of `plugin/snip.lua` depends on the `l` from the default-table.
The defaults are currently not exported (and I'm not sure we want to do so), but you can just copy [it]()

**Comment by leiserfg** [comment]
I think `snip_env` should extend the default and not replace it.

**Comment by hiberabyss** [comment]
> I think `snip_env` should extend the default and not replace it.

Yeah, I previously thought it worked this way. I could send a PR to do like this.

**Comment by L3MON4D3** [comment]
Mhmm, I don't know, some users don't like/use the `i`, `s`, single letter-acronyms for the nodes, and I wouldn't like to force those on them.
Maybe we should expose the defaults, then they can be `vim.tbl_extend`ed?
("and I'm not sure we want to do so" from my previous comment: I though that we would have much more breaking changes this way, but we should have a breaking change anyway when the defaults are changed, so it doesn't matter that they are explicitly exposed as well)

**Comment by L3MON4D3** [comment]
"force" is maybe a bit strong, more like have these globals defined when they are not used at all.

**Comment by leiserfg** [comment]
> "force" is maybe a bit strong, more like have these globals defined when they are not used at all.

They don't have to know :smile:

**Comment by L3MON4D3** [comment]
xD
Mhmm ahhhh yeaaaaahh... I'm not sure, on the one hand, we override default-stuff when a custom-table is assigned to `snip_env`, local variables will also override the defaults, so everything default is overridden by custom vars.
But I can see some variable being used without declaration, and maybe that variable shares a name with our default-stuff, and then, unexpectedly, there's not nil-error, but something completely different (granted, it'll probably be found right away as well, though I'm not certain of it)

I guess overriding with that table would be a bit more ergonomic, but some part of me can't stand it that there are some variables I'd have no way of getting rid of :sweat_smile:
